### PR TITLE
849 - Too much whitespace around titles and menu on mobile devices

### LINKED ISSUE #849: Too much whitespace around menus and titles on mobile devices
Example: 
68747470733a2f2f7777772e64726f70626f782e636f6d2f732f3333366572706c306639697a7268332f53637265656e73686f74253230323031362d30322d323925323030392e30302e34392e706e673f646c3d31

Seems to be a lot of wasted space here:

big gap before the "burger" menu
then an even bigger gap down to the "AgileVentures Events" title.
The title itself might look better in a smaller font so it fits on one line.

Note: The issue is not limited to this page, this is just an example

### PR DESCRIPTION
**Pull request by freeranger**
Closes #849
Deployed work in progress here: https://freeranger-wso-mobile.herokuapp.com

### CODE REVIEW

**Comment by freeranger** [comment]
@diraulo @tansaku I have put some screenshots on the issue itself of what I've done so far.
I'd like to push this to heroku so you can have a look but it's not working for me - I keep getting: 

>  Push rejected, failed to compile Ruby app

Are there any instructions anywhere on deploying WSO to a heroku instance?  
Is there any special config heroku needs?

Also there are conflicts with the changelog file again - I don't know how to resolve these.  I did a rebase to upstream/develop but it doesn't seem to have resolved them...

**Comment by diraulo** [comment]
@freeranger don't think we have instructions to deploy the app on Heroku anywhere. As far as I can recall, it was pretty straight forward for me the time I had to do it.

Regarding the conflict you're getting, I think you have to manually resolve the conflict in the file. The rebase will not do it automatically.

**Comment by freeranger** [comment]
@diraulo Hmm...I tried it a couple of times but no dice.  I saw some suggesting on the web about deleting the gemfile.lock, doing a bundle install, and pushing out again...but still no dice :/

Re: the conflict - I did resolve it manually and pushed -f back to origin but still got told there was a conflict :(

**Comment by freeranger** [comment]
@diraulo This is what I see locally:

> tony@Tonys-MacBook-Pro WebsiteOne (849_mobile_whitespace) $ git status
On branch 849_mobile_whitespace
Your branch is up-to-date with 'origin/849_mobile_whitespace'.
nothing to commit, working directory clean
tony@Tonys-MacBook-Pro WebsiteOne (849_mobile_whitespace) $ git merge upstream/develop
Already up-to-date.
tony@Tonys-MacBook-Pro WebsiteOne (849_mobile_whitespace) $ 

Yet the conflict still exists according to github...
You have contributor access to my repo btw if you want to take a look...

**Comment by tansaku** [comment]
@freeranger - you need to check that you have fetched the latest upstream/develop (I may need to update instructions) and then you might also need to push ...

**Comment by freeranger** [comment]
@tansaku hmm...Ok I did that and it worked but I don't undertand why I needed to when I did a `git merge upstream/develop`?
This was after trying `git rebase upstream/develop -i` which is what worked for me for the other PR's...

**Comment by tansaku** [comment]
@freeranger because git merge upstream/develop doesn't necessarily pull down any new code from the upstream

**Comment by freeranger** [comment]
@tansaku Thanks for the explanation...I'd have thought that's exactly what it does!
methinks I still have a lot to learn about git :/

**Comment by tansaku** [comment]
there's a lot to learn :-)

**Comment by freeranger** [comment]
@tansaku @diraulo This is largely done so please have a look and see what you think of the spacing and menu on smaller devices.
There are a couple of niggles I haven't gotten to the bottom of yet ( so any help appreciated)
- When you start to scroll, so the orange band disappears, you see a large white area which looks to be about the size the drop down menu would be if it was expanded:
So you go from this:
image

to this:

image

This is on Chrome - on safari, it is much less noticable as it seems to be the height of the normal dark menu bar, not the fully expanded menu.

- the burger menu "jumps" when you hover over it and will sometimes go dark and stay dark...
image

This is Chrome only - Safari seems fine

- On the members page (this happens currently too btw), the Members directory text disappears:
image
 
goes to

image


I know I've got a bit of tidying up to do with the PR and of course get these weird and wonderful issues sorted, but any feedback (And suggestions on the bugs) at this stage most appreciated.
It is deployed to: https://freeranger-wso-mobile.herokuapp.com

**Comment by tansaku** [comment]
@mtc2013 and I have replicated the issue in chrome mobile view

File: app/assets/stylesheets/layout/navbar.scss [hunk 1]
```diff
@@ -177,6 +181,12 @@ header#main_header {
     }
   }
 
+  .navbar-inverse, .navbar-toggle:hover {
+    margin-left:5px;
+    margin-top:5px;
```
Comment (mtc2013) [inline app/assets/stylesheets/layout/navbar.scss]: Making this margin-top be margin-top:8px; makes the burger jumping problem go away.

**Comment by tansaku** [comment]
The color change for the burger comes from the bootstrap class .navbar-toggle:focus, which gives changes to the darker color when you start scrolling with the touch point in the top navbar.

It looks to me like both issues can be resolved by removing the ".navbar-toggle:hover" class from navbar.scss - I think the burger is supposed to change color more consistently.

Have you been referring to http://getbootstrap.com/components/#navbar ?

I'm a little worried about extensive changes here based on code/css of uncertain nature underneath.

It might be good if you are very keen to keep working on this to create a responsive navbar from scratch and have a look at how that works by default, and then moving forward, being very careful about overriding styles that bootstrap is already using in certain patterns.

File: app/assets/stylesheets/layout/navbar.scss (hunk 1 above)
Comment (freeranger) [inline app/assets/stylesheets/layout/navbar.scss]: great, thanks!  I couldn't see margins changing in chrome dev tools when i hovered/moved away, so I couldn't identify what was causing it...

**Comment by freeranger** [comment]
@tansaku most of the changes I have made are not overriding the bootstrap settings.
The bootstrap page content does not help with our particular case because we are nonstandard in several ways:
1. The navbar is using the inverse colours when it is a desktop device but once we switch to mobile view, the burger is not inverse (as per https://github.com/AgileVentures/WebsiteOne/issues/849#issuecomment-196731109) but the actual menu still is
2. The menu is moving from the line below the "agileventures" heading to be part of that heading line when you switch to the mobile view and the burger
If you want to go "standard bootstrap" then the we need to revert to something like this, which I gave as an option before:
image 

Also please note that there are already a number of modifications to the navbar in the codebase: https://github.com/AgileVentures/WebsiteOne/blob/develop/app/assets/stylesheets/layout/navbar.scss
so even without my changes you aren't using the standard bootstrap styling.
This feels like a bigger issue as part of a restyle of the entire site - I'm not sure there's much value in doing it as part of this exercise to reduce whitespace to make the mobile view more usable.

I'd be happy to take out the menu styling completely and leave it like this:
image

and this issue really is just about whitespace around titles etc, which should be significantly better than it is now, and I can create another issue about removing all the custom overrides from the bootstrap menu if you like?

**Comment by freeranger** [comment]
Ok this is squashed and ready to go unless you want me to undo the burger me u changes...

**Comment by diraulo** [comment]
Sorry for not checking this earlier @freeranger I'll go through this thread today

**Comment by freeranger** [comment]
No problem @diraulo.  I mentioned you and @tansaku in the thread as the PM's to get some eyes on it since it is affecting the overall experience (on mobile).  
At this stage I guess it's either ready to go _or_ @tansaku will want me to pull the burger menu changes...but I'm away for the weekend so that wouldn't happen till next week anyway, so no rush.

**Comment by diraulo** [comment]
Hi @freeranger one concern with this is that we lost the logo on the navbar on the desktop version when scrolling the page. Was that intentional? 

before:
screenshot 2016-04-04 23 28 50

after:
screenshot 2016-04-04 23 28 29

**Comment by freeranger** [comment]
Hi @diraulo good spot!  No, that wasn't intended.   I'm not sure I like the current behaviour either tbh, where the orange strapline disappears when you scroll - it means that if you scroll down and then want to log in or sign up, you have to scroll back to the top again to do so...still, perhaps that's a conversation for another time.
I will look at the lost logo over the next couple of days.   
Is there anything else or do you think that's the only thing that needs fixing?

**Comment by diraulo** [comment]
@freeranger Just spacing between sections that need to be fixed. But I think that can be done as part of a separate PRs. There are already too many changes that are not so easy to track on this one, considering we have to test on different devices size :wink:

**Comment by freeranger** [comment]
Yes I guess it is meant to be about spacing around titles and menus so strictly speaking, I've gone too far with this...  I'll fix up the lost logo and we can get this one shipped hopefully!
It may be the rest is better left to be considered as part of thoughts on a wider redesign of the site...

**Comment by diraulo** [comment]
Indeed @freeranger. Just focus on restoring the logo on that navbar when scrolling then I'll pull your changes in.

**Comment by freeranger** [comment]
@diraulo Fixed the logo issue - should be good to go now, finally!

**Comment by diraulo** [comment]
Thanks for your work on this @freeranger. :shipit:

**Pull request merged**